### PR TITLE
Possibly improve std::variant readability

### PR DESCRIPTION
**Pull request by joshessman-llnl**
`std::variant` is kind of ugly to use without true pattern matching (hopefully in C++23), but `get_if` at least allows the type name to not be specified twice, which might be a little nicer to read.

There's some extra dereferencing involved with the coefficients, but #252 should clean that up.

### CODE REVIEW

File: src/physics/thermal_conduction.cpp [hunk 1]
```diff
@@ -21,18 +21,16 @@ ThermalConduction::ThermalConduction(int order, std::shared_ptr<mfem::ParMesh> m
...
-    auto dyn_params = std::get<DynamicSolverParameters>(params);
-    setTimestepper(dyn_params.timestepper);
+  else if (auto dyn_params = std::get_if<DynamicSolverParameters>(&params)) {
+    setTimestepper(dyn_params->timestepper);
     // Save these to initialize the DynamicConductionOperator later
-    dyn_M_params_ = dyn_params.M_params;
-    dyn_T_params_ = dyn_params.T_params;
+    dyn_M_params_ = dyn_params->M_params;
+    dyn_T_params_ = dyn_params->T_params;
   }
```
Comment (jamiebramwell) [inline src/physics/thermal_conduction.cpp:36]: Should we error if this block totally falls through or is that being too cautious?

File: src/physics/thermal_conduction.cpp [hunk 2]
```diff
@@ -21,18 +21,16 @@ ThermalConduction::ThermalConduction(int order, std::shared_ptr<mfem::ParMesh> m
   state_.push_back(temperature_);
 
   // If it's just the single set of params for a quasistatic K solve...
-  if (std::holds_alternative<LinearSolverParameters>(params)) {
-    auto K_params = std::get<LinearSolverParameters>(params);
-    K_inv_        = EquationSolver(mesh->GetComm(), K_params);
+  if (auto K_params = std::get_if<LinearSolverParameters>(&params)) {
```
Comment (jamiebramwell) [inline src/physics/thermal_conduction.cpp:24]: I like this syntax more.

**Review by jamiebramwell — APPROVED** [review]
I think this is a good improvement for readability.

**Comment by jamiebramwell** [comment]
LGTM

File: src/physics/thermal_conduction.cpp (hunk 1 above)
Comment (white238) [reply src/physics/thermal_conduction.cpp:36]: i like being cautious

**Review by white238 — APPROVED** [review]
Much better! thanks @joshessman-llnl !

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**